src/engine/UI/DisplayLayout.hpp

```diff
@@ -6,22 +6,21 @@
 
 #include "Element.hpp"
 #include "InteractiveDrawable.hpp"
+#include "OrderedCollection.hpp"
 
-namespace admirals {
-namespace UI {
+namespace admirals::UI {
 
 class DisplayLayout : public InteractiveDrawable {
 public:
     DisplayLayout();
 
-    void AddElement(std::shared_ptr<Element> element);
+    void AddElement(const std::shared_ptr<Element> element);
```

Comment:
no need to have const here since we pass by value, not reference

Reply:
const here means that the value of the Element (not the pointer) will not be changed by the function.

Reply:
If that is what we want, we would do `std::shared_ptr<const Element> element`, slapping the const before the shared pointer like `const std::shared_ptr<Element>` results in the shared pointe being constant (which doesn't matter in this case).